bridges/evm/contracts/example/DEX.sol

```diff
@@ -1,129 +1,102 @@
 pragma solidity 0.5.14;
 pragma experimental ABIEncoderV2;
 
-import {SafeMath} from "openzeppelin-solidity/contracts/math/SafeMath.sol";
-
-interface Bridge {
-    struct VerifyOracleDataResult {
-        bytes data;
-        bytes32 codeHash;
-        bytes params;
-    }
-    function relayAndVerify(bytes calldata data)
-        external
-        returns (VerifyOracleDataResult memory result);
-}
-
-contract DEX {
-    using SafeMath for uint256;
-
-    bytes32 public codeHash;
-
-    mapping(address => mapping(bytes => uint256)) _balances;
-
-    Bridge bridge = Bridge(0x3e1F8745E4088443350121075828F119075ef641);
-
-    constructor(bytes32 _codeHash) public {
-        codeHash = _codeHash;
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { IBridge } from "../IBridge.sol";
+
+contract DEX is Ownable {
+  using SafeMath for uint256;
+
+  bytes32 public codeHash;
+
+  mapping(address => mapping(bytes => uint256)) _balances;
+  mapping(string => bytes) public supportedTokens;
+
+  IBridge bridge;
+
+  constructor(IBridge _bridge, bytes32 _codeHash) public {
+    bridge = _bridge;
+    codeHash = _codeHash;
+    supportedTokens["ADA"] = hex"00000000";
+    supportedTokens["BAND"] = hex"00000001";
+    supportedTokens["BCH"] = hex"00000002";
+    supportedTokens["BNB"] = hex"00000003";
+    supportedTokens["BTC"] = hex"00000004";
+    supportedTokens["EOS"] = hex"00000005";
+    supportedTokens["ETC"] = hex"00000006";
+    supportedTokens["ETH"] = hex"00000007";
+    supportedTokens["LTC"] = hex"00000008";
+    supportedTokens["TRX"] = hex"00000009";
+    supportedTokens["XRP"] = hex"0000000A";
+  }
+
+  function isSupportedToken(string memory a) public view returns(bytes memory, bool) {
+    bytes memory key = supportedTokens[a];
+    return (key, key.length == 4);
```

Comment:
== 4 is a weird invariant. Let's do != 0 instead?

Reply:
done